com.musala.simple.students.db/src/main/java/com/musala/simple/students/db/MyMongoDatabase.java

```diff
@@ -0,0 +1,258 @@
+	 * 
+	 */
+	@Override
+	public void addStudent(Student student) {
+		Field[] fields = Student.class.getDeclaredFields();
+		Document newStudent = new Document();
+		for (Field field : fields) {
+			String fieldName = field.getName();
+			field.setAccessible(true);
+			try {
+				newStudent.append(fieldName, field.get(student));
+			} catch (IllegalArgumentException | IllegalAccessException e) {
+				logger.error(String.format("An error occured while trying to add student with name %s to the database.",
+						student.getName()));
+				logger.info("Stacktrace:", e);
+			}
+		}
+		try {
+			this.studentsCollection.insertOne(newStudent);
+		} catch (MongoWriteException e) {
+			if (e.getMessage().startsWith("E11000")) {
+				logger.error(String.format(
+						"Student %s with id %d can not be added to the Database. Student with the same id already exists\n",
+						student.getName(), student.getId()));
+			} else {
+				logger.error(String.format("Problem occured while trying to add %s with id %d to the database.\n",
+						student.getName(), student.getId()));
+				logger.error("Stacktrace:", e);
+			}
+		}
+	}
+
+	@Override
+	public void deleteStudentById(int studentId) {
+
+	}
+
+	/**
+	 * A MongoDb-specific implementation of the {@link Database#getStudentById()}
+	 * method. Searches the database for a document with an id property value equal
+	 * to the studentId param. If not found the method throws an exception. Else it
+	 * creates a student object with the required properties and returns it.
+	 * 
+	 * @param studentId
+	 *            the id to find a student in the database
+	 * @return a Student object with id, name, age and grade
+	 */
+	@Override
+	public Student getStudentById(int studentId) throws StudentNotFoundException {
+		Document query = new Document("id", studentId);
+		query = this.studentsCollection.find(query).first();
+
+		if (query == null) {
+			throw new StudentNotFoundException(ErrorMessage.STUDENT_NOT_EXISTS);
+		}
+
+		List list = new ArrayList(query.values());
+		int id = (int) list.get(1);
+		String name = (String) list.get(2);
+		int age = (int) list.get(3);
+		int grade = (int) list.get(4);
+
+		return new Student(id, name, age, grade);
+	}
+
+	/**
+	 * A MongoDb-specific implementation of the {@link Database#getAllStudentsArr()}
+	 * method. Retrieves the collection of {@link Document}s from the database, adds
+	 * the values of each document to a list and then assigns each value to the
+	 * corresponding property of a newly created Student objects. It then adds that
+	 * object to a List. After adding all the students to the list the method
+	 * returns an array from that list.
+	 * 
+	 * @return a Student[] array
+	 */
+	@Override
+	public Student[] getAllStudentsArr() {
+		List<Student> studentsList = new ArrayList<>();
+		try (MongoCursor<Document> cur = this.studentsCollection.find().iterator()) {
+			while (cur.hasNext()) {
+
+				Document doc = cur.next();
+
+				List list = new ArrayList(doc.values());
```

Comment:
The logic implemented between lines 200 and 204 is used in both `getStudentById()` and `getAllStudentsArr()` methods - extract it in an external method and call it in both places.